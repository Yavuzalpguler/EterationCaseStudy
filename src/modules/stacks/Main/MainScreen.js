import {
  View,
  Text,
  FlatList,
  SafeAreaView,
  TouchableOpacity,
  TextInput,
  RefreshControl,
} from 'react-native';
import React, {useState} from 'react';
import {useEffect} from 'react';
import {useDispatch} from 'react-redux';
import {getUserInfo, updateMainCustomState} from '../+store/Main/Main.action';

import {useSelector} from 'react-redux';
import UserItem from './components/UserItem';
import {
  heightPercentage,
  widthPercentage,
} from '../../../constants/StyleVariables';
import styles from './styles';
const MainScreen = props => {
  const dispatch = useDispatch();

  const userInfo = useSelector(state => state.main.userInfo);

  const handleUserItemClick = item => {
    props.navigation.navigate('MainDetail', {item});
  };

  const addUserButton = () => {
    return (
      <TouchableOpacity
        style={styles.addUserButtonWrapper}
        onPress={() => {
          props.navigation.navigate('AddUser');
        }}>
        <Text
          style={{
            fontSize: widthPercentage(10),

            lineHeight: widthPercentage(11),
            color: 'white',
          }}>
          {'\u002b'}
        </Text>
      </TouchableOpacity>
    );
  };

  const handleUserRemove = item => {
    dispatch(
      updateMainCustomState('userInfo', {
        data: userInfo.data.filter(user => user.id !== item.id),
      }),
    );
  };

  const handleUserShiftUp = item => {
    const index = userInfo.data.findIndex(user => user.id === item.id);
    if (index === 0) {
      return;
    }
    const temp = userInfo.data[index];
    userInfo.data[index] = userInfo.data[index - 1];
    userInfo.data[index - 1] = temp;
    dispatch(
      updateMainCustomState('userInfo', {
        data: userInfo.data,
      }),
    );
  };

  const handleUserShiftDown = item => {
    const index = userInfo.data.findIndex(user => user.id === item.id);
    if (index === userInfo.data.length - 1) {
      return;
    }
    const temp = userInfo.data[index];
    userInfo.data[index] = userInfo.data[index + 1];
    userInfo.data[index + 1] = temp;
    dispatch(
      updateMainCustomState('userInfo', {
        data: userInfo.data,
      }),
    );
  };

  const renderItem = ({item, index}) => {
    return (
      <>
        <UserItem
          item={item}
          index={index}
          handleUserItemClick={handleUserItemClick}
          handleUserRemove={handleUserRemove}
          handleUserShiftUp={handleUserShiftUp}
          handleUserShiftDown={handleUserShiftDown}
        />
      </>
    );
  };

  const renderPullToRefresh = () => {
    return (
      <View
        style={{
          alignSelf: 'center',
          paddingBottom: heightPercentage(2),
        }}>
        <Text style={{fontSize: widthPercentage(3), fontWeight: '700'}}>
          Pull to refresh the list
        </Text>
      </View>
    );
  };
  return (
    <SafeAreaView style={{flex: 1}}>
      {userInfo?.data?.length < 4 && renderPullToRefresh()}

      <FlatList
        data={userInfo?.data}
        numColumns={1}
        keyExtractor={(item, index) => index.toString()}
        contentContainerStyle={{paddingBottom: heightPercentage(10)}}
        renderItem={renderItem}
        refreshControl={
          <RefreshControl
            refreshing={userInfo?.loading}
            onRefresh={() => dispatch(getUserInfo())}
            colors={['red']}
            tintColor={'red'}
            progressBackgroundColor={'red'}
          />
        }
      />

      {addUserButton()}
    </SafeAreaView>
  );
};

export default MainScreen;
