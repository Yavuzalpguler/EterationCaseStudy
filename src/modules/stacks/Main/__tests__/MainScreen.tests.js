import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import MainScreen from '../MainScreen';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

const store = mockStore({
  main: {
    userInfo: {
      data: [
        {
          id: 1,
          name: 'Homer Simpson',
          age: 39,
          image: 'https://i.imgur.com/1TQOw9A.png',
        },
        {
          id: 2,
          name: 'Marge Simpson',
          age: 36,
          image: 'https://i.imgur.com/1TQOw9A.png',
        },
      ],
    },
  },
});
const navigation = {
  navigate: jest.fn(),
};

describe('MainScreen', () => {
  //clear mock store actions before each test
  beforeEach(() => {
    store.clearActions();
  });

  it('should render correctly', () => {
    const {toJSON} = render(
      <Provider store={store}>
        <MainScreen />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render correctly with empty data', () => {
    const store = mockStore({
      main: {
        userInfo: {
          data: [],
        },
      },
    });
    const {toJSON} = render(
      <Provider store={store}>
        <MainScreen />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render correctly with data', () => {
    const {toJSON} = render(
      <Provider store={store}>
        <MainScreen />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate to AddUser screen when add user button is pressed', () => {
    const navigation = {
      navigate: jest.fn(),
    };
    const {getByTestId} = render(
      <Provider store={store}>
        <MainScreen navigation={navigation} />
      </Provider>,
    );
    fireEvent.press(getByTestId('addUserButton'));
    expect(getByTestId('addUserButton')).toBeTruthy();
  });

  it('should navigate to AddUser screen when add button is pressed', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <MainScreen navigation={navigation} />
      </Provider>,
    );
    fireEvent.press(getByTestId('addUserButton'));
    expect(navigation.navigate).toHaveBeenCalledWith('AddUser');
  });

  it('should navigate to MainDetail screen when user item is pressed', () => {
    const {getByText} = render(
      <Provider store={store}>
        <MainScreen navigation={navigation} />
      </Provider>,
    );
    fireEvent.press(getByText('Homer Simpson'));
    expect(navigation.navigate).toHaveBeenCalledWith('MainDetail', {
      item: {
        id: 1,
        name: 'Homer Simpson',
        age: 39,
        image: 'https://i.imgur.com/1TQOw9A.png',
      },
    });
  });

  it('should remove user at index 0 when remove button is pressed', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <MainScreen />
      </Provider>,
    );

    fireEvent.press(getByTestId('RemoveButton0'));

    expect(store.getActions()).toEqual([
      {
        key: 'userInfo',
        type: '@UPDATE_MAIN_CUSTOM_STATE',
        value: {
          data: [
            {
              id: 2,
              name: 'Marge Simpson',
              age: 36,
              image: 'https://i.imgur.com/1TQOw9A.png',
            },
          ],
        },
      },
    ]);
  });

  it('should shift user up at index 1 when shift up button is pressed', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <MainScreen />
      </Provider>,
    );

    fireEvent.press(getByTestId('UpButton1'));

    expect(store.getActions()[0].value.data[0]).toEqual({
      id: 2,
      name: 'Marge Simpson',
      age: 36,
      image: 'https://i.imgur.com/1TQOw9A.png',
    });
  });

  it('should not shift user up at index 0 when shift up button is pressed', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <MainScreen />
      </Provider>,
    );

    fireEvent.press(getByTestId('UpButton0'));

    expect(store.getActions()).toEqual([]);
  });

  it('should shift user down at index 0 when shift down button is pressed', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <MainScreen />
      </Provider>,
    );
    fireEvent.press(getByTestId('DownButton0'));

    expect(store.getActions()[0].value.data[1]).toEqual({
      id: 2,
      name: 'Marge Simpson',
      age: 36,
      image: 'https://i.imgur.com/1TQOw9A.png',
    });
  });

  it('should not shift user down at index 1 when shift down button is pressed', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <MainScreen />
      </Provider>,
    );

    fireEvent.press(getByTestId('DownButton1'));

    expect(store.getActions()).toEqual([]);
  });
});

// Language: javascript
