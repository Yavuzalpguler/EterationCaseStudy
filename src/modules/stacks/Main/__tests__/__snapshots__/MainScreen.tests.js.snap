// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "paddingBottom": 26.68,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22.5,
          "fontWeight": "700",
        }
      }
    >
      Pull to refresh the list
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 133.4,
      }
    }
    data={
      Array [
        Object {
          "age": 39,
          "id": 1,
          "image": "https://i.imgur.com/1TQOw9A.png",
          "name": "Homer Simpson",
        },
        Object {
          "age": 36,
          "id": 2,
          "image": "https://i.imgur.com/1TQOw9A.png",
          "name": "Marge Simpson",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        colors={
          Array [
            "red",
          ]
        }
        onRefresh={[Function]}
        progressBackgroundColor="red"
        tintColor="red"
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(0,0,0,0.2)",
              "borderTopWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "height": 150,
              "justifyContent": "space-between",
              "opacity": 1,
              "padding": 15,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "flex": 0.2,
              }
            }
          >
            1
          </Text>
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "height": 112.5,
                  "width": 112.5,
                },
              ]
            }
          >
            <FastImageView
              defaultSource={null}
              resizeMode="contain"
              source={
                Object {
                  "priority": "normal",
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "color": "black",
                "flex": 5,
                "fontSize": 30,
                "textAlign": "left",
              }
            }
          >
            Homer Simpson
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "green",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="UpButton0"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                ↑
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="DownButton0"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                ↓
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="RemoveButton0"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                X
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(0,0,0,0.2)",
              "borderTopWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "height": 150,
              "justifyContent": "space-between",
              "opacity": 1,
              "padding": 15,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "flex": 0.2,
              }
            }
          >
            2
          </Text>
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "height": 112.5,
                  "width": 112.5,
                },
              ]
            }
          >
            <FastImageView
              defaultSource={null}
              resizeMode="contain"
              source={
                Object {
                  "priority": "normal",
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "color": "black",
                "flex": 5,
                "fontSize": 30,
                "textAlign": "left",
              }
            }
          >
            Marge Simpson
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "green",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="UpButton1"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                ↑
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="DownButton1"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                ↓
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="RemoveButton1"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                X
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "rgb(46,134,222)",
        "borderRadius": 75,
        "bottom": 75,
        "height": 112.5,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "width": 112.5,
      }
    }
    testID="addUserButton"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 75,
          "lineHeight": 82.5,
        }
      }
    >
      +
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`MainScreen should render correctly with data 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "paddingBottom": 26.68,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22.5,
          "fontWeight": "700",
        }
      }
    >
      Pull to refresh the list
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 133.4,
      }
    }
    data={
      Array [
        Object {
          "age": 39,
          "id": 1,
          "image": "https://i.imgur.com/1TQOw9A.png",
          "name": "Homer Simpson",
        },
        Object {
          "age": 36,
          "id": 2,
          "image": "https://i.imgur.com/1TQOw9A.png",
          "name": "Marge Simpson",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        colors={
          Array [
            "red",
          ]
        }
        onRefresh={[Function]}
        progressBackgroundColor="red"
        tintColor="red"
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(0,0,0,0.2)",
              "borderTopWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "height": 150,
              "justifyContent": "space-between",
              "opacity": 1,
              "padding": 15,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "flex": 0.2,
              }
            }
          >
            1
          </Text>
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "height": 112.5,
                  "width": 112.5,
                },
              ]
            }
          >
            <FastImageView
              defaultSource={null}
              resizeMode="contain"
              source={
                Object {
                  "priority": "normal",
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "color": "black",
                "flex": 5,
                "fontSize": 30,
                "textAlign": "left",
              }
            }
          >
            Homer Simpson
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "green",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="UpButton0"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                ↑
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="DownButton0"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                ↓
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="RemoveButton0"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                X
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(0,0,0,0.2)",
              "borderTopWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "height": 150,
              "justifyContent": "space-between",
              "opacity": 1,
              "padding": 15,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "flex": 0.2,
              }
            }
          >
            2
          </Text>
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "height": 112.5,
                  "width": 112.5,
                },
              ]
            }
          >
            <FastImageView
              defaultSource={null}
              resizeMode="contain"
              source={
                Object {
                  "priority": "normal",
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "color": "black",
                "flex": 5,
                "fontSize": 30,
                "textAlign": "left",
              }
            }
          >
            Marge Simpson
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "green",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="UpButton1"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                ↑
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="DownButton1"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                ↓
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "borderRadius": 22.5,
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 45,
                }
              }
              testID="RemoveButton1"
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 22.5,
                  }
                }
              >
                X
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "rgb(46,134,222)",
        "borderRadius": 75,
        "bottom": 75,
        "height": 112.5,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "width": 112.5,
      }
    }
    testID="addUserButton"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 75,
          "lineHeight": 82.5,
        }
      }
    >
      +
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`MainScreen should render correctly with empty data 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "paddingBottom": 26.68,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22.5,
          "fontWeight": "700",
        }
      }
    >
      Pull to refresh the list
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 133.4,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        colors={
          Array [
            "red",
          ]
        }
        onRefresh={[Function]}
        progressBackgroundColor="red"
        tintColor="red"
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "rgb(46,134,222)",
        "borderRadius": 75,
        "bottom": 75,
        "height": 112.5,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "width": 112.5,
      }
    }
    testID="addUserButton"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 75,
          "lineHeight": 82.5,
        }
      }
    >
      +
    </Text>
  </View>
</RCTSafeAreaView>
`;
